src/GafferUI/AuxiliaryConnectionGadget.cpp

```diff
@@ -0,0 +1,175 @@
+//
+//  Copyright (c) 2017, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+#include "GafferUI/AuxiliaryConnectionGadget.h"
+#include "GafferUI/GraphGadget.h"
+#include "GafferUI/NodeGadget.h"
+#include "GafferUI/Style.h"
+
+#include "Gaffer/Plug.h"
+#include "Gaffer/ScriptNode.h"
+
+using namespace GafferUI;
+using namespace Imath;
+
+AuxiliaryConnectionGadget::AuxiliaryConnectionGadget( const NodeGadget *srcGadget, const NodeGadget *dstGadget )
+	: Gadget(), m_srcGadget(srcGadget), m_dstGadget(dstGadget), m_toolTipValid(false)
+{
+}
+
+AuxiliaryConnectionGadget::~AuxiliaryConnectionGadget()
+{
+}
+
+Imath::Box3f AuxiliaryConnectionGadget::bound() const
+{
+	Box3f result;
+
+	result.extendBy( V3f( 0 ) * m_srcGadget->fullTransform() );
+	result.extendBy( V3f( 0 ) * m_dstGadget->fullTransform() );
+
+	return result;
+}
+
+void AuxiliaryConnectionGadget::doRenderLayer( Layer layer, const Style *style ) const
+{
+	if( layer != GraphLayer::Connections )
+	{
+		return;
+	}
+
+	// Compute endpoint of line
+	V3f source = V3f( 0 ) * m_srcGadget->fullTransform();
+	V3f destination = V3f( 0 ) * m_dstGadget->fullTransform();
+	V3f diff = source - destination;
+
+	// Put the direction indicator on an ellipse encompassing the target NodeGadget
```

Comment:
Yes, agreed. The annoying thing about that is that the Style will need to know about the bounds of the NodeGadgets involved, but that's probably ok. Depending on what we decide about AuxiliaryConnections to plugs with nodules, the Style might also have to know whether the endpoint is a NodeGadget center or a Nodule. Probably not worth discussing until the redesign of the core cocepts is done - especially because I remember you not being particularly keen on the elliptical indicator in general :) 
Let's assume we remove the ellipsis stuff, - maybe the LineSegment3f is just computed to hit the edge of the NodeGadget/Nodule and then we add an arrow right at the end. Probably solves all these problems.